### ISSUE TITLE
[url_launcher] Works without doing the configuration of README

### ISSUE THREAD

**Issue by juri180** [flutter/flutter#102630]
The configuration section in the README says that some configuration is necessary on iOS and Android.
- iOS: You've to add any URL schemes you're using to your Info.plist.
- Android: For API >=30 you've to add a `<queries>` element to the `AndroidManifest.xml`.

I tried to use the package for `https` and `http` without doing the configuration and everything worked. Tested with Android Emulator Pixel 3 API 30 and iPhone 13 iOS 15.2 on Simulator.

Why should I do the configuration, if it works also without? Or does it work only because of the Emulator/Simulator? It would be nice to have this in the documentation.

**Comment by maheshmnj** [comment]
Hi @juri180, Thanks for filing the issue.  I suspect that you are directly launching the URLs without querying the apps that can handle the URL. If you try the below code sample without adding queries to manifest, You will get the exception. 


```dart
  _launchUrl() async {
    final uri = Uri.parse('https://flutter.dev');
    if (await canLaunchUrl(uri)) {
      await launchUrl(uri);
    } else {
      throw 'Could not launch $uri';
    }
  }
```  

This exception is more common on recent Android versions. It is because the `canLaunch` looks for the URL handlers on the device and queries are required  to look for those handlers.

As stated in the doc comments of `canLaunchUrl`

> Checks whether the specified URL can be handled by some app installed on the device.

> Returns true if it is possible to verify that there is a handler available. A false return value can indicate either that there is no handler available, or that the application does not have permission to check. For example:

> On recent versions of Android and iOS, this will always return false unless the application has been configuration to allow querying the system for launch support. See the README for details.

**Comment by stuartmorgan** [comment]
There is a documentation issue here that I missed when updating the docs for the new API and reworked the discussion of `canLaunch`. We shouldn't be showing `http(s)` in the queries section, as there's almost no use case where someone would actually need to `canLaunch` check an `http(s)` URL on any of our supported platforms; browsers are pretty much always available. We also incorrectly comment the Android queries permission example as being about launching rather than querying.

**Comment by stuartmorgan** [comment]
> If you try the below code sample without adding queries to manifest, You will get the exception.

@maheshmnj Please don't demonstrate that sample going forward; it's unfortunate that the plugin ever suggested it in the first place, because it's actually an anti-pattern. `launch` will throw on failure already, so in that construction the `canLaunch` check is at best just pointless, and at worst causes issues (false negatives, unnecessary additions of query parameters).

I've already scrubbed it from the app-facing package, and I'll be updating the implementation packages to match shortly.

**Comment by juri180** [comment]
Thanks for your quick responses.

@maheshmnj I've tested the package with exactly the same code that you've provided and it worked for me without doing the configuration steps of the README on the test devices mentioned in the first message. I've also checked the Info.plist and AndroidManifest.xml to make sure that it's not already there.

@stuartmorgan Do I understand it correctly, that the Configuration section in the README for iOS and Android is indeed not necessary for simple `http(s)` launches?
And good to know, that this code snippet is not recommended. I've been using this implementation for a long time, because of the documentation.
